sources/packages/backend/apps/api/src/route-controllers/education-program/_tests_/e2e/education-program.aest.controller.deactivateProgram.e2e-spec.ts

```diff
@@ -0,0 +1,161 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeInstitution,
+  createFakeUser,
+} from "@sims/test-utils";
+import {
+  createTestingAppModule,
+  BEARER_AUTH_TYPE,
+  createFakeEducationProgram,
+  getAESTToken,
+  getAESTUser,
+  AESTGroups,
+} from "../../../../testHelpers";
+import * as request from "supertest";
+import { Institution, User } from "@sims/sims-db";
+
+describe("EducationProgramAESTController(e2e)-deactivateProgram", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let sharedInstitution: Institution;
+  let sharedUser: User;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sharedInstitution = createFakeInstitution();
+    await db.institution.save(sharedInstitution);
+    sharedUser = createFakeUser();
+    await db.user.save(sharedUser);
+  });
+
+  it("Should allow an education program to be deactivated when the program is active.", async () => {
+    // Arrange
+    const program = createFakeEducationProgram({
+      institution: sharedInstitution,
+      user: sharedUser,
+    });
+    await db.educationProgram.save(program);
+
+    const endpoint = `/aest/education-program/${program.id}/deactivate`;
+    const userToken = await getAESTToken(AESTGroups.BusinessAdministrators);
```

Comment:
Should we create a test for one of the non business administrator groups?

Reply:
We do not have requirements for it and even allowing administrators only is not a final requirement. I would not invest time in it.